src/main/java/spoon/support/reflect/declaration/CtNamedElementImpl.java

```diff
@@ -43,6 +49,22 @@ public String getSimpleName() {
 		if (factory instanceof FactoryImpl) {
 			simpleName = ((FactoryImpl) factory).dedup(simpleName);
 		}
+
+		if (this instanceof CtType && !simpleName.equals("")) {
```

Comment:
We should not change setSimpleName here. Instead we should change the CloneVisitorGenerator to call Refactor.rename().